Hookies/Game/Engine/GameEngine.swift

```diff
@@ -6,8 +6,278 @@
+        self.cannonSystem = CannonSystem(cannon: cannonSprite)
+
+        let finishingLineSprite = createFinishingLineSprite(from: finishingLine)
+        self.finishingLineSystem = FinishingLineSystem(finishingLine: finishingLineSprite)
+
+        setupMultiplayer()
+    }
+
+    // MARK: - Players
+
+    func setCurrentPlayer(id: String, position: CGPoint, image: String) {
+        let player = PlayerEntity()
+
+        let sprite = SpriteComponent(parent: player)
+        _ = spriteSystem.set(sprite: sprite, of: .player1, with: image, at: position)
+        _ = spriteSystem.setPhysicsBody(to: sprite, of: .player1, rectangleOf: sprite.node.size)
+        player.addComponent(sprite)
+
+        addCommonPlayerComponents(to: player)
+
+        currentPlayerId = id
+        currentPlayer = player
+
+        deadLockSystem = DeadlockSystem(sprite: sprite)
+        finishingLineSystem.add(player: sprite)
+    }
+
+    func addOtherPlayers(id: String, position: CGPoint, image: String) {
+        let otherPlayer = PlayerEntity()
+
+        let sprite = SpriteComponent(parent: otherPlayer)
+        let spriteType = getOtherPlayerSpriteType()
+        _ = spriteSystem.set(sprite: sprite, of: spriteType, with: image, at: position)
+        _ = spriteSystem.setPhysicsBody(to: sprite, of: spriteType, rectangleOf: sprite.node.size)
+        otherPlayer.addComponent(sprite)
+
+        addCommonPlayerComponents(to: otherPlayer)
+
+        otherPlayers[id] = otherPlayer
+
+        finishingLineSystem.add(player: sprite)
+    }
+
+    // MARK: - Bolts
+
+    private func initialiseBolts(_ bolts: [SKSpriteNode]) -> [SpriteComponent] {
+        var boltsSprite = [SpriteComponent]()
+
+        for bolt in bolts {
+            let boltEntity = BoltEntity()
+
+            let boltSprite = SpriteComponent(parent: boltEntity)
+            _ = spriteSystem.set(sprite: boltSprite, to: bolt)
+
+            // TODO: Check for moving and rotating bolt
+
+            boltEntity.addComponent(boltSprite)
+
+            boltsSprite.append(boltSprite)
+            self.bolts.append(boltEntity)
+        }
+
+        return boltsSprite
+    }
+
+    // MARK: - Cannon
+
+    private func createCannonSprite(from node: SKSpriteNode) -> SpriteComponent {
+        let sprite = SpriteComponent(parent: cannon)
+        _ = spriteSystem.set(sprite: sprite, to: node)
+
+        cannon.addComponent(sprite)
+
+        return sprite
+    }
+
+    // MARK: - Finishing Line
+
+    private func createFinishingLineSprite(from node: SKSpriteNode) -> SpriteComponent {
+        let sprite = SpriteComponent(parent: finishingLine)
+        _ = spriteSystem.set(sprite: sprite, to: node)
+        _ = spriteSystem.setPhysicsBody(to: sprite, of: .finishingLine, rectangleOf: sprite.node.size)
+
+        self.finishingLine.addComponent(sprite)
+
+        return sprite
+    }
+
+    // MARK: - Player helper methods
+
+    private func addCommonPlayerComponents(to player: PlayerEntity) {
```

Comment:
Consider naming it as `addPlayerComponents`. This should suffice to explain the method.